### PR TITLE
docs(samples): add export query results to Amazon S3 sample and IT

### PR DESCRIPTION
**Pull request by stephaniewang526**

### CODE REVIEW

**Review by pmakani — APPROVED** [review]

File: samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+package com.example.bigquery;
+
+// [START bigquery_omni_export_query_result_to_s3]
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.TableResult;
+
+// Sample to queries an external data source aws s3 using a permanent table
```
Comment (elharo) [inline samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java:26]: to query?
AWS S3 (caps)

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.io.PrintStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ExportQueryResultsToS3IT {
+  private final Logger log = Logger.getLogger(this.getClass().getName());
```
Comment (elharo) [inline samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:32]: log --> logger

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  private static final String OMNI_PROJECT_ID = requireEnvVar("OMNI_PROJECT_ID");
+  private static final String OMNI_DATASET_NAME = requireEnvVar("OMNI_DATASET_NAME");
+  private static final String OMNI_EXTERNAL_TABLE_NAME = requireEnvVar("OMNI_EXTERNAL_TABLE_NAME");
+  private static final String AWS_WRITE_CONNECTION_ID = requireEnvVar("AWS_WRITE_CONNECTION_ID");
+
+  private static String requireEnvVar(String varName) {
+    String value = System.getenv(varName);
+    assertNotNull(
+        "Environment variable " + varName + " is required to perform these tests.",
+        System.getenv(varName));
```
Comment (elharo) [inline samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:46]: value

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            "EXPORT DATA WITH CONNECTION `%s` OPTIONS(uri='%s', format='%s') "
+                + "AS SELECT * FROM %s.%s.%s WHERE name LIKE 'W%%'",
+            AWS_WRITE_CONNECTION_ID,
+            destinationUri,
+            format,
+            OMNI_PROJECT_ID,
+            OMNI_DATASET_NAME,
+            OMNI_EXTERNAL_TABLE_NAME);
+    ExportQueryResultsToS3.exportQueryResultsToS3(query);
+    assertThat(bout.toString()).contains("Query results exported to Amazon S3 successfully.");
```
Comment (elharo) [inline samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:89]: depends on default character set

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    originalPrintStream = System.out;
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown() {
+    // restores print statements in the original method
+    System.out.flush();
+    System.setOut(originalPrintStream);
+    log.log(Level.INFO, bout.toString());
```
Comment (elharo) [inline samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:71]: depends on default character set

File: samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  public static void main(String[] args) throws InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "MY_PROJECT_ID";
+    String datasetName = "MY_DATASET_NAME";
+    String externalTableName = "MY_EXTERNAL_TABLE_NAME";
+    // connectionName should be in the format of connection_region.connection_name. e.g.
+    // aws-us-east-1.s3-write-conn
+    String connectionName = "MY_CONNECTION_REGION.MY_CONNECTION_NAME";
+    String destinationUri = "s3://your-bucket-name/*";
```
Comment (lesv) [inline samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java:37]: why the * on dest?

**Review by lesv — APPROVED** [review]

File: samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java (hunk 1 above)
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java]: Comment updated.

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java (hunk 2 above)
Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:32]: Ack

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java (hunk 4 above)
Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:89]: Ack

File: samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java (hunk 6 above)
Comment (stephaniewang526) [reply samples/snippets/src/main/java/com/example/bigquery/ExportQueryResultsToS3.java:41]: Comment added!

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java (hunk 5 above)
Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:71]: Ack

File: samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java (hunk 3 above)
Comment (stephaniewang526) [reply samples/snippets/src/test/java/com/example/bigquery/ExportQueryResultsToS3IT.java:46]: Ack

**Pull request merged**